fix "--resolver global " needs network

ppzzppz opened the pull request:
issue: https://github.com/commercialhaskell/stack/issues/5103


* [ x ] Any changes that could be relevant to users have been recorded in the ChangeLog.md
* [ x ] The documentation has been updated, if necessary.

Test case:
```
stack test   673 examples, 0 failures
stack --resolver global   is ok
stack --resolver nightly is ok
stack --resolver lts-14   is ok
stack --resolver lts-14.13 is ok
```

qrilka commented on src/Stack/Config.hs:
```diff
@@ -146,7 +146,7 @@ makeConcreteResolver
 makeConcreteResolver (ARResolver r) = pure r
 makeConcreteResolver ar = do
-    snapshots <- getSnapshots
+    -- Close #5103
```
This doesn't seem to be a meaningful comment here

qrilka commented on src/Stack/Config.hs:
```diff
@@ -157,16 +157,19 @@ makeConcreteResolver ar = do
-                     in return $ ltsSnapshotLocation x y
+            ARLatestLTS -> do
+                snapshots <- getSnapshots
+                let r | IntMap.null $ snapshotsLts snapshots = throwString "No LTS releases found"
```
Minor nitpicking but will it not look simpler if `if` were used here? Then `let` will not be required and that trailing `r` as well...

ppzzppz commented:
Received. Thanks for review.

Could changeLog be written under "Changes since v2.1.3.1"?


Such as the following：

```
**Changes since v2.1.3.1**

Release notes:

Major changes:

Behavior changes:

Other enhancements:

* Remove warning for using Stack with GHC 8.8 and Cabal 3.0.
* `--resolver global` doesn't retrieve snapshots list from the internet beause doesn't need it

Bug fixes:

```

qrilka commented:
Surely it's since v2.1.3.1 the question is whether it's an enhancement or a bug fix, I'm a bit more inclined to see it as a bug fix
